src/modules/paymentProcessor/PP_Everclear_CrossChain_v1.sol

```diff
@@ -0,0 +1,427 @@
+            intent_.initiator,
+            intent_.receiver,
+            intent_.inputAsset,
+            intent_.outputAsset,
+            intent_.maxFee,
+            intent_.origin,
+            intent_.nonce,
+            intent_.timestamp,
+            intent_.ttl,
+            intent_.amount,
+            intent_.destinations,
+            intent_.data
+        );
+    }
+
+    /// @notice Transfer the token for the payment order into the payment processor.
+    /// @dev    The tokens must be transferred to the payment processor first because
+    ///         the bridge contract will later call this contract to execute the
+    ///         cross-chain transfer.
+    /// @param  order_ The payment order details.
+    /// @param  client_ The payment client address.
+    function _transferTokenAndApproveToBridge(
+        IERC20PaymentClientBase_v2.PaymentOrder memory order_,
+        address client_
+    ) internal virtual {
+        IERC20(order_.paymentToken).safeTransferFrom(
+            client_, address(this), order_.amount
+        );
+        // Update the amount paid on the payment client side.
+        IERC20PaymentClientBase_v2(client_).amountPaid(
+            order_.paymentToken, order_.amount
+        );
+
+        IERC20(order_.paymentToken).approve(
+            address(_everClearSpoke), order_.amount
+        );
+    }
+
+    /// @notice Creates a new cross-chain intent for payment transfer.
+    /// @param  order_ The payment order details.
+    /// @return intentId_ ID of the created intent.
+    function _createCrossChainIntent(
+        IERC20PaymentClientBase_v2.PaymentOrder memory order_
+    )
+        internal
+        virtual
+        returns (bytes32 intentId_, IEverclear.Intent memory intent_)
+    {
+        // Get the max fee and TTL from the flags and data.
+        (uint24 maxFee, uint48 ttl) =
+            _getEverclearMaxFeeAndTTL(order_.flags, order_.data);
+        uint32[] memory destinations = new uint32[](1);
+        destinations[0] = uint32(order_.targetChainId);
+        // Properly capture both return values
+        (intentId_, intent_) = _everClearSpoke.newIntent(
+            destinations,
+            order_.recipient,
+            order_.paymentToken,
+            order_.paymentToken,
+            order_.amount,
+            maxFee,
+            ttl,
+            ""
+        );
+        return (intentId_, intent_);
+    }
+
+    /// @notice Gets the Everclear max fee and TTL from the data.
+    /// @param  flags_ The flags indicating which data elements are present.
+    /// @param  data_ The data to get the max fee and TTL from.
+    /// @return maxFee_ The max fee.
+    /// @return ttl_ The TTL.
+    function _getEverclearMaxFeeAndTTL(bytes32 flags_, bytes32[] memory data_)
+        internal
+        pure
+        virtual
+        returns (uint24 maxFee_, uint48 ttl_)
+    {
+        uint relativeIndex = 0;
+
+        for (uint8 i = 0; i <= FLAG_TTL; ++i) {
```

Comment:
Wouldn't the retrieval of the values in the data array be wrong if e.g. we have a payment order that has flags and an array of 5 long? So in this case, where the wrong payment order might have been sent to this payment processor, the PP will take the first and second array entry as values. I believe your version which you implemented in the streaming PP would prevent this?